packages/fxa-settings/src/styles/custom/UnitRow.scss

```diff
@@ -0,0 +1,59 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+.unit-row {
+  @apply flex flex-wrap px-4 pt-7 pb-8;
+
+  &-header {
+    @apply w-full mb-1;
+  }
+
+  &-header-value {
+    @apply w-full;
+  }
```

Comment:
Using `&-` here to one layer felt right to help the styles here be a bit more concise, two seemed to make it less readable. We could choose not to ever use `&-` because this does make it less searchable.

Reply:
I'm torn about these re-used classes in general. I can see why they were created, and how they are handy, but at a certain point it starts to look like we're writing custom CSS (if only to use `@apply`), and an added drawback IMO is needing a `custom/` dir. Could these classes simply exist in component classNames? There will be duplication of class names (as is the spirit of Tailwind), but it's more flexible in the long run and alleviates the prior concerns.

What do you think?

Reply:
As I've continued working on this and looking at the wires, I see the need for a `cta-default` class (what `.cta` is now), `cta-default-lg` for modals, `cta-primary`, and `cta-warning` or whatever we want to call the red one. If we _don't_ do it this way, there's going to be _a lot_ of places where we're going to need to add/remove classes when something needs to be updated. I feel like there's greater risk of missing something when we have a repeated set of classes. I did this originally with `UnitRow` because if we needed to change the padding on something there, well, it would be _very_ easy to update it in `UnitRow` and forget to update it in `UnitRowWithImage`.

Check out [this video from the docs](https://tailwindcss.com/course/composing-utilities-with-apply/#app). They actually do exactly this so I still think it's very Tailwindy. 😎

Reply:
Right, I do understand why you've set it up this way, and I agree it is handy. I guess my thinking goes back to concerns that when we start applying global styles to re-used components we could be painting ourselves into a corner if an instance of the component needs to be overridden in such a way that competes with existing styles. Is this approach not the same as applying base or concatenated Tailwind classes to shared components instead of passing all of them in as props for each instance?

Reply:
If it competes with existing styles, only the styles the component needs should be set on that component.

> Is this approach not the same as applying base or concatenated Tailwind classes to shared components instead of passing all of them in as props for each instance?

Well, check out `cta-default`, I'm referencing that class in 3 places in this PR (and made a `cta-default-lg` since the modal one had similarities but not the exact styles, and noticed we'd need the same style for the "separate page" flows). I don't think every single instance of what I've got for `cta-default` should be pulled into its own component, that would feel complicated just for the sake of not doing it this way, and I think that'd really complicate `UnitRow` and `UnitRowWithImage` if we want to keep those the same. If you want to poke around the branch I'm all ears on other solutions, this one just feels like the cleanest, and Tailwind docs do recommend it when you've got a long set of classes used in more than one spot.

Reply:
BTW, after we reach consensus here, not sure if I'll have time this sprint but I want to write up a quick "follow these conventions" note in the README; the one you wrote on SVGs and what we've already got in there is handy. :)

Reply:
> If it competes with existing styles, only the styles the component needs should be set on that component.

I think that's where potential problems lie, though? Let's say Foo component is styled with a `foo` class that applies Tailwind classes to set properties A through F. And this works for everything, until we need an instance of Foo with a different C value. Does that mean we stop using `foo` (is this possible if the class is baked into? Or apply the additional className and have classes that introduce competing values?

All things considered, though, I recognize this set of classes is for a really far-reaching set of elements, so I'm less concerned about this in general, but I agree that we should put up some examples and establish if this is considered standard practice, or if it's an exception. Thank you for the debating here :)

Reply:
Yeah, you definitely have a point here - depending on the situation, maybe that one differing style could be pulled out and applied, so right now we have:

```
.unit-row {
  @apply flex flex-wrap px-4 pt-7 pb-8;
}
```
Say we need to use this class, but we need a `pb-4` instead. Our options are:
1. Apply a `unit-row pb-4` combination class, overwriting `pb-8`
2. Applying `flex flex-wrap px-4 pt-7 pb-4` directly on the one-off
3. Remove `pb-8` from `unit-row`, finding all instances of where it's used, and apply `unit-row pb-8` there, and `unit-row pb-4` on the one-off instance

I think it could depend on the circumstance. I could potentially see cases where all 3 might be OK, but if you'd like to avoid 1 entirely, we can make note of that.